File: packages/extract/test/fixtures/polyglot/csharp/Job.cs
using System;
using System.Threading;

enum State { Idle, Running, Done, Failed, Retrying }

class Job {
    private State state = State.Idle;
    private Timer timer;

    public void Start() {
        if (state != State.Idle) return;
        state = State.Running;
        timer = new Timer(_ => Console.WriteLine("tick"), null, 0, 1000);
    }

    public void Finish() {
        if (state == State.Running) state = State.Done;
    }

    public void Fail() {
        state = State.Failed;
    }

    public string Describe() {
        switch (state) {
            case State.Idle: return "waiting";
            case State.Running: return "busy";
            case State.Done: return "finished";
        }
        return "";
    }
}

File: packages/extract/test/fixtures/polyglot/java/src/main/java/app/Job.java
package app;

import java.io.FileReader;
import java.io.IOException;

enum State { IDLE, RUNNING, DONE, FAILED, RETRYING }

public class Job {
    private State state = State.IDLE;

    public void start() {
        if (state != State.IDLE) return;
        state = State.RUNNING;
    }

    public void finish() {
        if (this.state == State.RUNNING) {
            this.state = State.DONE;
        }
    }

    public void fail() {
        state = State.FAILED;
    }

    public String describe() {
        switch (state) {
            case IDLE: return "waiting";
            case RUNNING: return "busy";
            case DONE: return "finished";
        }
        return "";
    }

    public String firstLine(String path) throws IOException {
        FileReader reader = new FileReader(path);
        int c = reader.read();
        reader.close();
        return String.valueOf(c);
    }
}
